isotovideo

```diff
@@ -335,14 +296,9 @@ sub _calc_check_delta {
     if ($last_check_seconds) {
         $delta = tv_interval([$last_check_seconds, $last_check_microseconds], [gettimeofday]);
     }
-    if ($delta > 0) {
-        # sleep the remains of one second
-        $timeout = 1 - $delta;
-        $timeout = 0 if $timeout < 0;
-    }
-    else {
-        $timeout = 0;
-    }
+    # sleep the remains of one second if $delta > 0
+    my $timeout = $delta > 0 ? 1 - $delta : 0;
```

Comment:
`my $timeout = max(0, min(1 - $delta, 1))` ?

or even add a helper, `clamp(0, 1 - $delta, 1)` - I think this construct is used in several places.

Reply:
In my opinion the max/min version is even harder to read. Adding a helper would make sense, but only if we give it a meaningful name. (I don't think `clamp` is meaningful enough.)